test/rulesets/Base/base.jl

```diff
@@ -174,4 +174,21 @@
         test_frule(Base.literal_pow, ^, 3.5, Val(3))
         test_rrule(Base.literal_pow, ^, 3.5, Val(3))
     end
+
+    @testset "Float conversions" begin
+        for f in (float, Float32, Float64, AbstractFloat)
+            test_frule(f, 1.2; rtol=1.0e-3, atol=1.0e-3)
+            test_frule(f, 1⊢1.7; rtol=1.0e-3, atol=1.0e-3)
+            test_rrule(f, 1.2; rtol=1.0e-3, atol=1.0e-3)
+
+            # test_rrule doesn't like integers, so test that case manually
+            let y = rand(Int), x = randn()
+                @test rrule(f, y)[2](x)[2] == x
+            end
+        end
+
+        # Make sure that we didn't accidentally define a rule for all DataTypes
+        @test frule(NoRules, 1.0) === nothing
+        @test rrule(NoRules, 1.0) === nothing
```

Comment:
I think this should be removed from this PR.
And a seperate PR should add just this, as its own testset in its own file

Reply:
Maybe, but it's the rules in this PR that accidentally define it if you're not careful.

Reply:
Correct.
I think we should have it, just not here

Reply:
Alright. I would prefer to merge as is though, so we can fix Diffractor and then move things around in a follow up.